src/models/__tests__/profile.test.ts

```diff
@@ -184,147 +185,151 @@ describe("Profile codec", () => {
+    expect(containerMock.items.create).toHaveBeenCalledWith(
+      expect.objectContaining({
+        kind: "INewProfile",
+        fiscalCode: aProfileWithDifferentEmail.fiscalCode,
+        email: aProfileWithDifferentEmail.email
+      }),
+      expect.anything()
     );
 
-    expect(clientMock.createDocument).toHaveBeenCalledTimes(1);
-    expect(clientMock.createDocument.mock.calls[0][1].kind).toBeUndefined();
-    expect(clientMock.createDocument.mock.calls[0][2]).toHaveProperty(
-      "partitionKey",
-      aFiscalCode
-    );
-    expect(isRight(result)).toBeTruthy();
-    if (isRight(result)) {
-      expect(result.value.isSome()).toBeTruthy();
-      if (isSome(result.value)) {
-        const updatedProfile = result.value.value;
-        expect(updatedProfile.fiscalCode).toEqual(aRetrievedProfile.fiscalCode);
-        expect(updatedProfile.id).toEqual(`${aFiscalCode}-${"0".repeat(15)}1`);
-        expect(updatedProfile.version).toEqual(1);
-        expect(updatedProfile.email).toEqual("new@example.com");
-      }
+    expect(E.isRight(result)).toBeTruthy();
+    if (E.isRight(result)) {
+      const updatedProfile = result.right;
+      expect(updatedProfile.fiscalCode).toEqual(aRetrievedProfile.fiscalCode);
+      expect(updatedProfile.id).toEqual(`${aFiscalCode}-${"0".repeat(15)}1`);
+      expect(updatedProfile.version).toEqual(1);
+      expect(updatedProfile.email).toEqual("new@example.com");
     }
   });
 
-  it("should reject the promise in case of error (read)", async () => {
-    const clientMock: any = {
-      createDocument: jest.fn(),
-      readDocument: jest.fn((_, __, cb) => cb("error"))
-    };
+  it("should get COSMOS_ERROR_RESPONSE in case of create error", async () => {
+    const containerMock = ({
+      items: {
+        create: jest.fn().mockReturnValue(Promise.reject())
+      }
+    } as unknown) as Container;
 
-    const model = new ProfileModel(clientMock, profilesCollectionUrl);
+    const model = new ProfileModel(containerMock);
 
-    const result = await model.update(aFiscalCode, aFiscalCode, o => o);
+    const result = await model.update(aRetrievedProfile)();
 
-    expect(clientMock.readDocument).toHaveBeenCalledTimes(1);
-    expect(clientMock.createDocument).not.toHaveBeenCalled();
+    expect(containerMock.items.create).toHaveBeenCalledTimes(1);
 
-    expect(isLeft(result)).toBeTruthy();
-    if (isLeft(result)) {
-      expect(result.value).toEqual("error");
+    expect(E.isLeft(result)).toBeTruthy();
+    if (E.isLeft(result)) {
+      expect(result.left.kind).toEqual("COSMOS_ERROR_RESPONSE");
     }
   });
 
-  it("should reject the promise in case of error (create)", async () => {
-    const clientMock: any = {
-      createDocument: jest.fn((_, __, ___, cb) => cb("error")),
-      readDocument: jest.fn((_, __, cb) => cb(undefined, aRetrievedProfile))
-    };
+  it("should get COSMOS_EMPTY_RESPONSE in case of empty resource", async () => {
```

Comment:
can't this happen with the creation?

Reply:
in some weird scenarios it can appen, ill add a test for it. thanks